server/src/main/java/io/crate/metadata/NodeContext.java

```diff
@@ -22,21 +22,24 @@
 package io.crate.metadata;
 
 import io.crate.role.Roles;
-import org.elasticsearch.common.inject.Inject;
-import org.elasticsearch.common.inject.Singleton;
+import java.util.function.Function;
 
-@Singleton
 public class NodeContext {
 
     private final Functions functions;
     private final long serverStartTimeInMs;
     private final Roles roles;
 
-    @Inject
-    public NodeContext(Functions functions, Roles roles) {
+    private final Function<NodeContext, Schemas> createSchemas;
+
+    private volatile Schemas schemas;
+
+    public NodeContext(Functions functions, Roles roles, Function<NodeContext, Schemas> createSchemas) {
         this.functions = functions;
-        this.serverStartTimeInMs = SystemClock.currentInstant().toEpochMilli();;
+        this.serverStartTimeInMs = SystemClock.currentInstant().toEpochMilli();
         this.roles = roles;
+        this.createSchemas = createSchemas;
+        schemas();
```

Comment:
If this works eagerly couldn't we make it final and remove the lazy initialization logic?